src/java/org/apache/cassandra/auth/AuthConfig.java

```diff
@@ -56,11 +56,11 @@ public static void applyAuth()
         // work with PasswordAuthenticator, so log a message if some other authenticator
         // is in use and non-default values are detected
         if (!(authenticator instanceof PasswordAuthenticator)
-            && (conf.credentials_update_interval_in_ms != -1
-                || conf.credentials_validity_in_ms != 2000
+            && (conf.credentials_update_interval != null
```

Comment:
Is it possible for `credentials_update_interval` to be null?

Reply:
I think I meant to have` && (conf.credentials_update_interval.toMilliseconds() != 0)`. I had similar change I did it in DatabaseDescriptor but seems I forgot here and our tests didn't catch it.... :-( Correcting, good catch! Thanks!

Confirmed, found it in the DatabaseDescriptor - https://github.com/ekaterinadimitrova2/cassandra/blob/CASSANDRA-15234-take2/src/java/org/apache/cassandra/config/DatabaseDescriptor.java#L1306

Now I think the "-1" change is a different and a valuable discussion to have. In the initial version I remember having "-1" considered for backward compatibility but if I remember correctly I was told that 0 and -1 would do the same so negative ns/ms/s/etc can be considered as a bug so I corrected accordingly. Unfortunately, this was Slack discussion more than a year ago and my history does not have it anymore. Also, many things happened since then so I think it is worth it to revise the topic.

I just looked at the docs again... - I don't find anything mentioned about -1 explicitly.
What we have: 

```
credentials_validity_in_ms


Validity period for credentials cache. This cache is tightly coupled to the provided PasswordAuthenticator implementation of IAuthenticator. If another IAuthenticator implementation is configured, this cache will not be automatically used and so the following settings will have no effect. Please note, credentials are cached in their encrypted form, so while activating this cache may reduce the number of queries made to the underlying table, it may not bring a significant reduction in the latency of individual authentication attempts. Defaults to 2000, set to 0 to disable credentials caching.
Default Value: 2000
credentials_update_interval_in_ms

This option is commented out by default.
Refresh interval for credentials cache (if enabled). After this interval, cache entries become eligible for refresh. Upon next access, an async reload is scheduled and the old value returned until it completes. If credentials_validity_in_ms is non-zero, then this must be also. Defaults to the same value as credentials_validity_in_ms.
Default Value: 2000


```

Two key points for me are: 
`set to 0 to disable credentials caching.` and `If credentials_validity_in_ms is non-zero, then this must be also.`.

Do you think from your experience that "-1" usage can be some kind of tribal knowledge maybe? Or is there anything else that I am missing? 

There are three properties with default value -1 in Config - `permissions_update_interval_in_ms, roles_update_interval_in_ms, credentials_update_interval_in_ms`.

Reply:
As long as we make sure that this check continues to be in sync w/ the default value, I think we're fine. If that default value is the "boxed zero", and it also indicates "disabled", then I think `conf.credentials_update_interval.toMilliseconds() != 0` works. If I'm reading correctly, I think that means we won't change the default behavior at all.